File: Python/shopping.py



toys = [ 20, 100, 10, 22, 18, 11 ]


def how_many_toys(input, k):
    summation = 0
    count = 0

    for i in input:
        if  summation + i <= k:
            summation += i
            count += 1
    return count



if __name__ == "__main__":

    k = int(input("How much money do you have to spend? "))
    toys.sort()
    print(toys)
    toy_count = how_many_toys(toys, k)

    print(f"You can buy {toy_count} items.")

File: C/shopping.cpp
#include <bits/stdc++.h>

using namespace std;

int max_toys(int cost[], int N, int k){

	int sum = 0;
	int count = 0;


	for (int i = 0; i < N; i++){

		if ( sum + cost[i] <= k){
			sum = sum + cost[i];
			count++;
		}

	}
	return count;
}

int main(){

	int k;
	cout << "How much money do you have to spend? " << endl;
	cin >> k;

	//int k = 50;
	int cost[] = {10,11,12,13,20,25,50};
	int N = sizeof(cost) / sizeof(cost[0]);

	cout << "You can buy " << max_toys(cost, N, k) << " items with $" << k << "." << endl;
	return 0;

}
